src/fdb5/database/Catalogue.h

```diff
@@ -114,16 +115,16 @@ class CatalogueReader {
 public:
     virtual DbStats stats() const = 0;
     virtual bool axis(const std::string& keyword, eckit::StringSet& s) const = 0;
-    virtual bool retrieve(const Key& key, Field& field) const = 0;
+    virtual bool retrieve(const ApiKey& key, Field& field) const = 0;
 };
 
 
 class CatalogueWriter {
 public:
     virtual const Index& currentIndex() = 0;
-    virtual void archive(const Key& key, std::unique_ptr<FieldLocation> fieldLocation) = 0;
+    virtual void archive(const ApiKey& key, std::unique_ptr<FieldLocation> fieldLocation) = 0;
```

Comment:
These feel peculiar - why is ApiKey making it as far down as archive() and retrieve() - I would expect the ApiKey to become canonicalised, and the Canonical Key to get to here?